File: Cap 12/Topico 12.7/Polimorfismo/Ad-hoc/Coercao/Exemplo 1/polimcoer1.cpp
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 12 - Topico 12.7
 * 
 * Programa ..: POLIFORMISMO AD-HOC DE COERCAO
 * 
 */

#include <iostream>
using namespace std;

void ESCREVANUM(int NUMERO)
{
  cout << "Numero inteiro: " << NUMERO << endl;
}

void ESCREVANUM(double NUMERO)
{
  cout << "Numero real: " << NUMERO << endl;
}

int main(void)
{

  int VALOR1 = 10;
  double VALOR2 = 3.14159;

  ESCREVANUM(VALOR1); // Chama funcao ESCREVANUM(int)
  ESCREVANUM(VALOR2); // Chama funcao ESCREVANUM(double)

  return 0;
}

File: Cap 12/12.7 - Poliformismo/12.7.4 - Poliformismo Ad-hoc de Coercao/Exemplo 1/Simulado/polimcoer1.php
<?php

/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 12 - Topico 12.7
 * 
 * Programa ..: POLIFORMISMO AD-HOC DE COERCAO (SIMULADO)
 * 
 */

function ESCREVANUM($NUMERO) {
  if (is_int($NUMERO)) {
    echo "Numero inteiro: " . $NUMERO . PHP_EOL;
  } elseif (is_float($NUMERO)) {
    echo "Numero real: " . $NUMERO . PHP_EOL;
  }
}

$VALOR1 = 10;
$VALOR2 = 3.14159;

ESCREVANUM($VALOR1); // Chama função ESCREVANUM(int)
ESCREVANUM($VALOR2); // Chama função ESCREVANUM(double)

?>
